### PR TITLE
HHH-18170 Subquery randomly generating wrong SQL due to duplicate alias

### PR DESCRIPTION
**Pull request by mpela81**
https://hibernate.atlassian.net/browse/HHH-18170

Possible implementation replaces various `System.nanoTime()` calls, used to generate an unique identifier, with a counter.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/spi/SqmCreationHelper.java [hunk 1]
```diff
@@ -35,10 +39,14 @@ public static NavigablePath buildSubNavigablePath(NavigablePath lhs, String base
 		return lhs.append( base, determineAlias( alias ) );
 	}
 
+	public static String getUniqueId() {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/spi/SqmCreationHelper.java:42]:
```suggestion
	public static String acquireUniqueAlias() {
```

**Comment by beikov** [comment]
The change looks good, I'd just prefer a different name for the method.

File: hibernate-core/src/main/java/org/hibernate/query/sqm/spi/SqmCreationHelper.java (hunk 1 above)
Comment (mpela81) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/spi/SqmCreationHelper.java]: done

**Review by beikov — APPROVED** [review]

**Comment by beikov** [comment]
Thanks!

**Pull request merged**